shell/browser/api/atom_api_web_contents.cc

```diff
@@ -1868,8 +1879,23 @@ void WebContents::Print(gin_helper::Arguments* args) {
   auto* rfh = focused_frame && focused_frame->HasSelection()
                   ? focused_frame
                   : web_contents()->GetMainFrame();
-  print_view_manager->PrintNow(rfh, silent, std::move(settings),
-                               std::move(callback));
+
+  base::PostTaskAndReplyWithResult(
+      FROM_HERE,
+      {base::ThreadPool(), base::MayBlock(), base::TaskPriority::USER_BLOCKING},
+      base::BindOnce(&GetDefaultPrinterAsync),
+      base::BindOnce(
+          [=](base::string16* printer_name, base::Value print_settings,
+              printing::CompletionCallback print_callback,
+              base::string16 default_printer) {
+            if (printer_name->empty())
+              *printer_name = default_printer;
+            print_settings.SetStringKey(printing::kSettingDeviceName,
+                                        *printer_name);
+            print_view_manager->PrintNow(rfh, silent, std::move(print_settings),
+                                         std::move(print_callback));
+          },
+          &device_name, std::move(settings), std::move(callback)));
```

Comment:
with `&device_name` here it looks like we're passing in the address of a stack local that's going out-of-scope by the time this lambda is invoked?